primer.md

```diff
@@ -128,6 +129,34 @@ The following will not be part of the specification:
 * Language-specific runtime APIs
 * Selecting a single identity/access control system
 
+
+## Architecture
+
+The CloudEvents specification set defines four different kinds of protocol
+elements that form a layered architecture model.
+
+1. The [base specification](spec.md) defines an abstract information model and
+   associated rules for what constitutes CloudEvent.
+2. The [extensions](./extensions) add use-case specific and potentially
```

Comment:
From an architectural standpoint, I agree we should introduce extensions separately.

However, I don't think this link should go to `./extensions`, but to `./spec.md#extension-attributes`. The spec defines what extension attributes are. The extensions folder contains a few "standardized" extensions, but from an architectural point of view, it is irrelevant whether an extension is defined in the repo, or completely custom.